### PR TITLE
FAI-8219 - Bitbucket Server source updates

### PR DESCRIPTION
**Pull request by cjwooo**
## Description

- Add bucketing options
- Add option to use different endpoint for fetching users in project. Unfortunately this endpoint requires admin permissions, but it might be a lot faster performing than the default "filter all users for project permissions" endpoint.

## Type of change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change

## Related issues

> Fix [#1]() 

## Migration notes

> Describe migration notes if any

## Extra info

> Add any additional information

### CODE REVIEW

File: sources/bitbucket-server-source/src/bitbucket-server/index.ts [hunk 1]
```diff
@@ -493,33 +516,49 @@ export class BitbucketServer {
...
       throw new VError(
         innerError(err),
-        `Error fetching users for project ${project}`
+        `Error fetching users for project ${projectKey}`
       );
     }
   }
+
+  bucket(repoFullName: string): number {
+    const md5 = createHmac('md5', 'farosai/airbyte-bitbucket-server-source');
```
Comment (tovbinm) [inline sources/bitbucket-server-source/src/bitbucket-server/index.ts:557]: what is the `'farosai/airbyte-bitbucket-server-source'` for?

**Review by tovbinm — CHANGES_REQUESTED** [review]
see comments

File: sources/bitbucket-server-source/src/bitbucket-server/index.ts (hunk 1 above)
Comment (cjwooo) [reply sources/bitbucket-server-source/src/bitbucket-server/index.ts:557]: HMAC key for generating the hash.

**Comment by cjwooo** [comment]
@tovbinm I'm merging this PR to get it released for our deadline. Will address any feedback in subsequent PRs.

**Pull request merged**